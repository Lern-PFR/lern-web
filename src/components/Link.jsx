import React, { useState } from 'react';
import PropTypes from 'prop-types';

const STATUS =  {
    HOVERED: 'hovered',
    NORMAL: 'normal'
};

const Link = ({ page, children }) => {

    const [status, setStatus] = useState(STATUS.NORMAL);

    return (
        <a
            className={status}
            href={page || '#'}
            onMouseEnter={() => setStatus(STATUS.HOVERED)}
            onMouseLeave={() => setStatus(STATUS.NORMAL)}
        >
            { children && children}
        </a>
    );
}

Link.propTypes = {
    page: PropTypes.string,
    children: PropTypes.any,
};

export default Link;