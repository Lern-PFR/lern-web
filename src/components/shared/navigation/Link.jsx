import PropTypes from 'prop-types';
import { Link as ReactRouterLink } from 'react-router-dom';
import styled from 'styled-components';
import { color } from 'styled-system';

const StyledLink = styled(ReactRouterLink)(
	{
		cursor: 'inherit',
		color: 'inherit',
		fontFamily: 'inherit',
		fontSize: 'inherit',
		fontWeight: 'inherit',
		textDecoration: 'inherit',
	},
	color,
);

/**
 * @name Link
 * @description Renders a link to an internal URL, by providing keys path to the route.
 *
 * @author Yann Hodiesne
 *
 * @param {string}	to			key of the route to target with this Link
 * @param {bool}	[disabled]	Whether the link is disabled or not. If so, the onClick event is aborted.
 */
const Link = ({ to, disabled, ...props }) => (
	<StyledLink to={to} {...(disabled ? { onClick: (e) => e.preventDefault() } : {})} {...props} />
);

Link.propTypes = {
	to: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.object,
		PropTypes.func,
	]),
	disabled: PropTypes.bool,
};

Link.defaultProps = {
	disabled: false,
};

Link.defaultProps = {
	to: '',
};

export default Link;
