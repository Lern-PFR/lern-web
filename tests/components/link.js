import React from 'react';
import Link from '../../src/components/Link';
import { shallow } from 'enzyme';

describe('Link component', () => {
    it('should match the snapshot', () => {
        // Arrange
        const sut = shallow(<Link />)

        // Act

        // Assert
        expect(sut).toMatchSnapshot();
    });

    it('should change its css class when hovered', () => {
        // Arrange       
        const sut = shallow(<Link />)

        // Act
        sut.find('a').simulate('mouseenter');

        // Assert
        expect(sut.find('a').hasClass('hovered')).toBe(true);
    });
});