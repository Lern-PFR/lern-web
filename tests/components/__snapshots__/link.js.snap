// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component should match the snapshot 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;
